File: bindings/wasm/examples/owned_objects.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import {
  Address,
  GraphQlClient,
  ObjectFilter,
  initAsync,
} from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();
const address = Address.zero();
const objectsPage = await client.objects(ObjectFilter.new({ owner: address }));
console.log(`Owned objects(${objectsPage.data.length}):`);
for (const obj of objectsPage.data) {
  console.log(obj.id().toHex());
}

File: crates/iota-sdk/examples/owned_objects.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use eyre::Result;
use iota_sdk::{
    graphql_client::{Client, pagination::PaginationFilter, query_types::ObjectFilter},
    types::Address,
};

#[tokio::main]
async fn main() -> Result<()> {
    let client = Client::new_testnet();

    let address = Address::ZERO;
    let owned_objects_page = client
        .objects(
            Some(ObjectFilter {
                owner: Some(address),
                ..Default::default()
            }),
            PaginationFilter::default(),
        )
        .await?;
    println!("Owned objects ({}):", owned_objects_page.data.len());
    for obj in owned_objects_page.data {
        println!("{}", obj.id());
    }

    Ok(())
}
